pkg/kernel/signal_once.go

```diff
@@ -0,0 +1,48 @@
+package kernel
+
+import (
+	"sync"
+)
+
+//go:generate mockery -name=SignalOnce
+type SignalOnce interface {
```

Comment:
👍 for this abstraction. I think it was already being used like that in `kernel.go`. There was a channel that was just being closed. But having this type makes the intention of that channel much more clear.